### DISCUSSION TITLE
redact annots can not remove overlap contents

### DISCUSSION THREAD

**Discussion by marc-chin** [pymupdf/PyMuPDF#1477] [Looking for help] [answered]
hi there,

i tried to use the below code to remove the contents in my pdf document,  add redact annot can mask the content(drawing), but once i use adobe PDF pro DC to edit the document(Touch up tool ), after delete the redact annot, the drawing appeal again, which mean the redact annot can not remove the contents completely, right? 

rect=fitz.Rect(2,2,200,27)
page.wrap_contents()
page.add_redact_annot(rect,text='xxxxxx',fill=None)
page.apply_redactions(images=1)

doc.scrub(redactions=True,redact_images=2)
doc.save('removelogo.pdf',garbage=3)

how can remove the contents in the document completely?

**Comment by JorjMcKie** [comment, marked as answer]
You are right:
The redaction logic only removes
* links that overlap a redact rect
* text characters whose (somewhat reduced) bbox overlaps a redact rect
* images ...
    - either completely if overlapping a redact rect **_and_** `fitz.PDF_REDACT_IMAGE_REMOVE` is specified,
    - or not at all if `fitz.PDF_REDACT_IMAGE_NONE` is specified,
    - or the overlapping part if `fitz.PDF_REDACT_IMAGE_PIXELS` is specified.

Other overlapping objects, e.g. drawings, remain untouched - there is no way out.
